src/http/multipart.rs

```diff
@@ -5,11 +5,23 @@ use reqwest::multipart::{Form, Part};
 use crate::builder::CreateAttachment;
 use crate::internal::prelude::*;
 
+impl CreateAttachment {
+    fn into_part(self) -> Result<Part> {
+        let mut part = Part::bytes(self.data);
+        part = guess_mime_str(part, &self.filename)?;
+        part = part.file_name(self.filename);
+        Ok(part)
+    }
+}
+
 /// Holder for multipart body. Contains files, multipart fields, and payload_json for creating
 /// requests with attachments.
 #[derive(Clone, Debug)]
 pub struct Multipart {
-    pub files: Vec<CreateAttachment>,
+    /// Files that are sent with the form data as individual uploads.
+    pub upload_file: Option<CreateAttachment>,
+    /// Files that are sent with the form data as message attachments.
+    pub attachment_files: Option<Vec<CreateAttachment>>,
```

Comment:
Do we still want to combine these into a single enum, since they seem to be mutually exclusive? Would remove the need for `CreateAttachment::into_part` as the part could be calculated inline using a `match`.

Reply:
Sure we can, I don't particularly care. I frankly forgot that idea was a thing while creating this follow-up PR

Reply:
I experimented a bit with names locally, and I kind of like this definition:
```rust
pub enum MultipartUpload {
    /// A file sent with the form data as an individual upload. For example, a sticker.
    File(CreateAttachment),
    /// Files sent with the form as message attachments.
    Attachments(Vec<CreateAttachment>),
}
```
Originally, if you remember, the proposed enum name was simply `Files`, but that seemed too generic to me. And, on its own the `File` variant may also seem generic but when written as `MultipartUpload::File`, I think it's pretty understandable. It only would occur in a single place anyways.

Reply:
I like it too! Nice idea